tensorflow_probability/python/mcmc/transformed_kernel_test.py

```diff
@@ -308,6 +328,89 @@ def test_is_calibrated(self):
+        new_kernel.bootstrap_results(5.),
+    ])
+    self.assertNear(9., pkr_one.transformed_state, err=1e-6)
+    self.assertNear(5., pkr_two.transformed_state, err=1e-6)
+
+  def test_bijector_valid_transform_then_step(self):
+    new_kernel = transform_then_step(tfb.Exp())
+    pkr_one, pkr_two = self.evaluate([
+        new_kernel.bootstrap_results(2.),
+        new_kernel.bootstrap_results(9.),
+    ])
+    self.assertNear(np.log(2.),
+                    pkr_one.inner_results.transformed_state,
+                    err=1e-6)
+    self.assertNear(np.log(9.),
+                    pkr_two.inner_results.transformed_state,
+                    err=1e-6)
+
+  def test_bijector_valid_step_then_transform(self):
+    new_kernel = step_then_transform(tfb.Exp())
+    pkr_one, pkr_two = self.evaluate([
+        new_kernel.bootstrap_results(2.),
+        new_kernel.bootstrap_results(9.),
+    ])
+    self.assertNear(np.log(2.), pkr_one.transformed_state, err=1e-6)
+    self.assertNear(np.log(9.), pkr_two.transformed_state, err=1e-6)
+
+  def test_step_size_changed(self):
+    target_dist = tfd.MultivariateNormalDiag(loc=[0., 0.], scale_diag=[1., 10.])
+    # `hmc_kernel`'s step size is far from optimal
+    hmc_kernel = tfp.mcmc.HamiltonianMonteCarlo(
+        target_log_prob_fn=target_dist.log_prob,
+        num_leapfrog_steps=27,
+        step_size=10)
+    step_adaptation_kernel = tfp.mcmc.SimpleStepSizeAdaptation(
+        inner_kernel=hmc_kernel,
+        num_adaptation_steps=9)
+    trans_kernel = tfp.mcmc.TransformedTransitionKernel(
+        inner_kernel=step_adaptation_kernel,
+        bijector=tfb.Exp()
+    )
+    kernel_results = trans_kernel.inner_kernel.bootstrap_results(tf.zeros(2))
+    for _ in range(2):
+      _, kernel_results = trans_kernel.inner_kernel.one_step(tf.zeros(2),
+                                                             kernel_results)
+    adapted_step_size = self.evaluate(
+        kernel_results.inner_results.accepted_results.step_size)
+    # the step size for `FakeInnerKernel` is 10
+    self.assertFalse(adapted_step_size == 10)
+
+  def test_deeply_nested(self):
+    trans_kernel = tfp.mcmc.TransformedTransitionKernel(
+        inner_kernel=FakeInnerKernel(target_log_prob_fn=fake_target_log_prob),
+        bijector=tfb.Exp())
+    step_kernel = tfp.mcmc.SimpleStepSizeAdaptation(
+        inner_kernel=trans_kernel,
+        num_adaptation_steps=9)
+    double_step_kernel = tfp.mcmc.SimpleStepSizeAdaptation(
+        inner_kernel=step_kernel,
+        num_adaptation_steps=9)
+    pkr_one, pkr_two = self.evaluate([
+        double_step_kernel.bootstrap_results(2.),
+        double_step_kernel.bootstrap_results(9.),
+    ])
+    self.assertNear(np.log(2.),
+                    pkr_one.inner_results.inner_results.transformed_state,
+                    err=1e-6)
+    self.assertNear(np.log(9.),
+                    pkr_two.inner_results.inner_results.transformed_state,
+                    err=1e-6)
 
```

Comment:
Can you add another test to make sure `TransformedTransitionKernel` correctly nests with itself?  Use two different bijectors to make sure you are testing the order in which they are composed.

Reply:
Added.